app/src/main/java/org/fossasia/susi/ai/skills/settings/ChatSettingsFragment.kt

```diff
@@ -112,9 +112,13 @@ class ChatSettingsFragment : PreferenceFragmentCompat(), ISettingsView {
         if (!utilModel.isLoggedIn()) {
             displayEmail.title = "Not logged in"
             displayEmail.isEnabled = true
+            speechOutput.isEnabled = false
+            speechAlways.isEnabled = false
```

Comment:
Why?

Reply:
sir, this was the flaw that was stated in the issue. User should be allowed to change voice setting only if he is logged in. The code base written below also showed similar approach.

Reply:
Why should he only be allowed when he's logged in?

Reply:
Sir, the issue created by @ravi-kishan stated that thing only. Also, when I analyzed the code base. It pretended to do the same. It had used the anonymity function to check wether the user is signed in or not.